src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
+                    res.push_back(std::make_pair("Contract Type", "Foundation Poll"));
+
+                else
+                    res.push_back(std::make_pair("Contract Type", "Poll"));
+
+                std::string sPollTitle = ExtractXML(msg, "<TITLE>", "</TITLE>");
+                std::replace(sPollTitle.begin(), sPollTitle.end(), '_', ' ');
+                std::string sPollDays = ExtractXML(msg, "<DAYS>", "</DAYS>");
+                std::string sPollQuestion = ExtractXML(msg, "<QUESTION>", "</QUESTION>");
+                std::string sPollAnswers = ExtractXML(msg, "<ANSWERS>", "</ANSWERS>");
+                std::string sPollShareType = ExtractXML(msg, "<SHARETYPE>", "</SHARETYPE>");
+                std::string sPollUrl = ExtractXML(msg, "<URL>", "</URL");
+                std::string sPollExpiration = ExtractXML(msg, "<EXPIRATION>", "</EXPIRATION>");
+                std::string sPollCommit = ExtractXML(msg, "<MA>", "</MA>");
+
+                if (sPollCommit == "A")
+                    res.push_back(std::make_pair("Poll Commit", "Add"));
+
+                else if (sPollCommit == "D")
+                    res.push_back(std::make_pair("Poll Commit", "Delete"));
+
+                else
+                    res.push_back(std::make_pair("Poll Commit", "Unknown"));
+
+                res.push_back(std::make_pair("Poll Title", sPollTitle));
+                res.push_back(std::make_pair("Poll Question", sPollQuestion));
+                std::replace(sPollAnswers.begin(), sPollAnswers.end(), ';', ',');
+                res.push_back(std::make_pair("Poll Answers", sPollAnswers));
+                sPollShareType = GetShareType(std::stod(sPollShareType));
+                res.push_back(std::make_pair("Poll Share Type", sPollShareType));
+                res.push_back(std::make_pair("Poll URL", sPollUrl));
+                res.push_back(std::make_pair("Poll Duration", sPollDays + " days"));
+                res.push_back(std::make_pair("Poll Expires", TimestampToHRDate(std::stod(sPollExpiration))));
+            }
+
+            else if (sMessageType == "vote")
+            {
+                res.push_back(std::make_pair("Contract Type", "Vote"));
+
+                std::string sVoteTitle = ExtractXML(msg, "<TITLE>", "</TITLE>");
+                std::string sVoteShareType = GetPollXMLElementByPollTitle(sVoteTitle, "<SHARETYPE>", "</SHARETYPE>");
+                sVoteShareType = GetShareType(std::stod(sVoteShareType));
+                std::replace(sVoteTitle.begin(), sVoteTitle.end(), '_', ' ');
+                std::string sVoteAnswer = ExtractXML(msg, "<ANSWER>", "</ANSWER>");
+                std::replace(sVoteAnswer.begin(), sVoteAnswer.end(), ';', ',');
+
+                res.push_back(std::make_pair("Poll Title", sVoteTitle));
+                res.push_back(std::make_pair("Poll Answer", sVoteAnswer));
+                res.push_back(std::make_pair("Poll Share Type", sVoteShareType));
```

Comment:
Could you perhaps add Vote weight?

Reply:
ya i am going to add that. just got to pull the magnitude form the vote tx and money supply from the block staked at that time. ive already half coded it

Reply:
also theres another issue i got to address to do this. 

`double GetMoneySupplyFactor()
{

        StructCPID structcpid = mvNetwork["NETWORK"];
        double TotalCPIDS = mvMagnitudes.size();
        double AvgMagnitude = structcpid.NetworkAvgMagnitude;
        double TotalNetworkMagnitude = TotalCPIDS*AvgMagnitude;
        if (TotalNetworkMagnitude < 100) TotalNetworkMagnitude=100;
        double MoneySupply = DoubleFromAmount(pindexBest->nMoneySupply);
        double Factor = (MoneySupply/TotalNetworkMagnitude+.01);
        return Factor;

}`

as you see here the deciding factor at the time of the vote takes quite a few things into account so i'd need to go back and try to pull the most accurate values possible

Reply:
i could maybe just do a Moneysupply at the time of tx (from the block it was in) and just use 115000 for TotalNetworkMagnitude . but that would be truely just an estimate.

Reply:
may also try pulling superblock data from around that time :) to get that avg mag and total cpids.

Reply:
this one is becoming a headache. last step i got to do its app cache filter for vote and find the number of votes on the poll etc. will do that part tmr

Reply:
Can't you just extract the vote weight from the transaction?
I am worried that retrieving historic data and scanning appcache might not only be tedious to implement, but also slow.

Reply:
vote weight is not store in vote. just the balance and magnitude. vote weight. at the time has many factors such as money supply factor which uses the avg mag from the time of the vote of the network and the total participant count that has mag.

Reply:
GetMoneySupplyFactor uses the current network averages not the averages at time of vote. i'm allow people to see there old vote txs as well as well as look back in future if they need to. i'll make it as efficient as possible

Reply:
In that case, just showing what is in the transaction might be better.

Reply:
i figured it out and made it work :) and its fast with appcache as well :)

Reply:
-- complete no need for appcache lol (thx ravon) voting part is done now to correct the rest of labels etc and see about translations

Reply:
--complete can grab voting weight up to ~6 months back